### PR TITLE
Feature/Experimental/ Add more date ranges to pick from

### LINKED ISSUE #2903: Replace toggle with select in assistant component
Replace `gf-toggle` with `mat-select` in `AssistantComponent` and replace the date range options:

```
public dateRangeOptions = [
  { label: $localize`Today`, value: '1d' },
  { label: $localize`Week to date` + ' (' + $localize`WTD` + ')', value: 'wtd' },
  { label: $localize`Month to date` + ' (' + $localize`MTD` + ')', value: 'mtd' },
  { label: $localize`Year to date` + ' (' + $localize`YTD` + ')', value: 'ytd' },
  { label: $localize`1Y`, value: '1y' },
  { label: $localize`5Y`, value: '5y' },
  { label: $localize`Max`, value: 'max' }
];
```

### PR DESCRIPTION
**Pull request by meuter**
- Add `Week to date` and `Month to date` to the list of possible date ranges to pick from.
- Instead of using toggle buttons, the date range is picked using a drop down selection.

Fixes #2903

# Todo

- [x] replace `dateRangeOptions` with extra values in `AssistantComponent`
- [x] replace `gf-toggle` by `mat-select`
- [x] by default have the entry matching `user?.settings?.dateRange` selected.

### CODE REVIEW

File: libs/ui/src/lib/assistant/assistant.component.ts [hunk 1]
```diff
@@ -92,7 +92,24 @@ export class AssistantComponent implements OnChanges, OnDestroy, OnInit {
 
   public static readonly SEARCH_RESULTS_DEFAULT_LIMIT = 5;
 
-  public dateRangeOptions = ToggleComponent.DEFAULT_DATE_RANGE_OPTIONS;
+  public dateRangeOptions = [
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.component.ts:95]: public **readonly** dateRangeOptions...

File: libs/ui/src/lib/assistant/assistant.component.ts [hunk 2]
```diff
@@ -92,7 +92,24 @@ export class AssistantComponent implements OnChanges, OnDestroy, OnInit {
 
   public static readonly SEARCH_RESULTS_DEFAULT_LIMIT = 5;
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.component.ts:94]: Add `public dateRangeFormControl = new FormControl<string>(undefined);`

File: libs/ui/src/lib/assistant/assistant.component.ts
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.component.ts:1]: Add
```
this.dateRangeFormControl.setValue(this.user?.settings?.dateRange ?? null);
```
at the beginning of `public ngOnChanges()`

Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.component.ts:1]: Remove `ToggleComponent` import

File: libs/ui/src/lib/assistant/assistant.html [hunk 3]
```diff
@@ -102,12 +102,18 @@
...
         ></ng-template
       >
-      <div class="d-flex justify-content-center py-3">
-        <gf-toggle
-          [defaultValue]="user?.settings?.dateRange"
-          [options]="dateRangeOptions"
-          (change)="onChangeDateRange($event.value)"
-        ></gf-toggle>
+      <div class="p-3">
+        <mat-form-field>
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.html:106]: `<mat-form-field appearance="outline" class="w-100 without-hint">`

File: libs/ui/src/lib/assistant/assistant.html [hunk 4]
```diff
@@ -102,12 +102,18 @@
...
+          <mat-label>Date Range</mat-label>
+          <mat-select
+            placeholder="No date range selected"
+            (selectionChange)="onChangeDateRange($event.value)"
+          >
+            @for (range of dateRangeOptions; track range) {
+            <mat-option [value]="range.value"> {{ range.label }} </mat-option>
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.html:113]: Remove spaces before and after `{{ range.label }}`

File: libs/ui/src/lib/assistant/assistant.module.ts [hunk 5]
```diff
@@ -19,6 +22,8 @@ import { AssistantComponent } from './assistant.component';
     FormsModule,
     GfAssistantListItemModule,
     GfToggleModule,
+    MatFormFieldModule,
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.module.ts:25]: Move down to keep the alphabetical order

File: libs/ui/src/lib/assistant/assistant.module.ts [hunk 6]
```diff
@@ -19,6 +22,8 @@ import { AssistantComponent } from './assistant.component';
...
+    MatSelectModule,
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.module.ts:26]: Move down to keep the alphabetical order

**Review by dtslvr — CHANGES_REQUESTED** [review]
You can achieve the data binding with a form control. Please find the hints in my comments.

File: libs/ui/src/lib/assistant/assistant.html [hunk 7]
```diff
@@ -102,12 +102,18 @@
...
       >
-      <div class="d-flex justify-content-center py-3">
-        <gf-toggle
-          [defaultValue]="user?.settings?.dateRange"
-          [options]="dateRangeOptions"
-          (change)="onChangeDateRange($event.value)"
-        ></gf-toggle>
+      <div class="p-3">
+        <mat-form-field>
+          <mat-label>Date Range</mat-label>
```
Comment (dtslvr) [inline libs/ui/src/lib/assistant/assistant.html:107]: Remove, I think it's clear by the tab label

**Comment by meuter** [comment]
Oh wow, thanks for the detailed code review! I'll make the necessary changes and update the PR.

**Review by dtslvr — APPROVED** [review]
Thank you @meuter for this cool feature including the video! 😎

**Comment by dtslvr** [comment]
> There is one thing I am not 100% sure about though: as you'll see in the video, when I select "Week to date", I only get one sample point on the graph (since the video was recorded on Monday). I would have expected a graph looking like the "Today" view. It may be link to the way the backend handle the start of week:
> 
> ```
> startOfWeek(new Date().setHours(0, 0, 0, 0), { weekStartsOn: 1 })
> ```

Hmm, for `1d` one day is subtracted. I have also added it now for `wtd` and it seems to work. But do you think we need it as well for `mtd`?

**Comment by meuter** [comment]
Thanks for the guidance throughout the PR. I learned a lot ;-)

> Hmm, for 1d one day is subtracted. I have also added it now for wtd and it seems to work. But do you think we need it as well for mtd?

I think it makes sense indeed. Otherwise on the first day of the month we will have the same one point plot.

**Pull request merged**